File: pydss/general/NamedObjectImpl.py
from dss.general.NamedObject import (NamedObject,)
# from java.util.UUID import (UUID,)


class NamedObjectImpl(object, NamedObject):
    pathName = None
    localName = None
    displayName = None
    uuid = None

    def __init__(self, className):
        super(NamedObjectImpl, self)()
        self.pathName = className
        self.localName = ''
        self.displayName = ''
        self.uuid = None

    def finalize(self):
        if self.uuid is not None:
            self.uuid = None

    def getDisplayName(self):
        if self.displayName == '':
            return self.pathName + '_' + self.localName
        else:
            return self.displayName

    def setDisplayName(self, value):
        self.displayName = value

    def getQualifiedName(self):
        return self.pathName + '.' + self.localName

    def getUUID(self):
        if self.uuid is None:
            self.uuid = UUID.randomUUID()
        return self.uuid

    def setUUID(self, value):
        # if (pGuid == null) {}
        self.uuid = value

    def getID(self):
        return str(self.getUUID())

    def getCIM_ID(self):
        return self.UUIDToCIMString(self.getUUID())

    def getDSSClassName(self):
        return self.pathName

    def setDSSClassName(self, value):
        self.pathName = value

    def getLocalName(self):
        return self.localName

    def setLocalName(self, value):
        self.localName = value

    def UUIDToCIMString(self, uUID):
        s = str(uUID)
        return s[1:-2]

File: core/jdss/src/main/java/com/epri/dss/general/impl/NamedObjectImpl.java
/**
 * Copyright (C) 2009 Electric Power Research Institute, Inc.
 * Copyright (C) 2011 Richard Lincoln
 *
 * All rights reserved.
 */

package com.epri.dss.general.impl;

import java.util.UUID;

import com.epri.dss.general.NamedObject;

public class NamedObjectImpl implements NamedObject {

	/**
	 * Path name, or class name for DSS.
	 */
	private String pathName;

	/**
	 * LocalName is unique within a class.
	 */
	private String localName;

	/**
	 * For optional display, does not have to be unique.
	 */
	private String displayName;

	private UUID uuid;


	public NamedObjectImpl(String className) {
		super();
		pathName = className;
		localName = "";
		displayName = "";
		uuid = null;
	}

	public String getDisplayName() {
		if (this.displayName == "") {
			return pathName + "_" + localName;
		} else {
			return displayName;
		}
	}

	public void setDisplayName(String value) {
		displayName = value;
	}

	public String getQualifiedName() {
		return pathName + "." + localName;
	}

	private UUID getUUID() {
		if (uuid == null)
			uuid = UUID.randomUUID();
		return uuid;
	}

	public void setUUID(UUID value) {
		//if (pGuid == null) {}
		uuid = value;
	}

	public String getID() {
		return getUUID().toString();
	}

	public String getCIM_ID() {
		return UUIDToCIMString(getUUID());
	}


	public String getDSSClassName() {
		return pathName;
	}

	public void setDSSClassName(String value) {
		pathName = value;
	}

	public String getLocalName() {
		return localName;
	}

	public void setLocalName(String value) {
		localName = value;
	}


	public String UUIDToCIMString(UUID uUID) {
		String s;
		s = uUID.toString();
		return s.substring(1, s.length() - 2);
	}

}
